From 120d18e311947ce5c66f1b90cd8ba4eb548028b1 Mon Sep 17 00:00:00 2001
From: Charlie Sharpsteen <source@sharpsteen.net>
Date: Sat, 11 May 2019 13:30:15 -0700
Subject: [PATCH 08/13] Update facter to build on debian armhf

This commit updates the facter component to build on the debian-9-armhf
and debian-10-armhf platforms using Debian's cmake and to link against
Debians's Boost and YAML-CPP libraries.
---
 configs/components/facter.rb | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/configs/components/facter.rb b/configs/components/facter.rb
index 7cd20d91..c28c29d4 100644
--- a/configs/components/facter.rb
+++ b/configs/components/facter.rb
@@ -120,6 +120,23 @@ component "facter" do |pkg, settings, platform|
     boost_static_flag = "-DBOOST_STATIC=OFF"
     special_flags += "-DCMAKE_CXX_FLAGS='#{settings[:cflags]}' -DENABLE_CXX_WERROR=OFF"
     yamlcpp_static_flag = "-DYAMLCPP_STATIC=OFF"
+  elsif platform.is_cross_compiled_linux? && platform.name =~ /debian-(?:9|10)/
+    # Use system cmake and libraries
+    cmake = '/usr/bin/cmake'
+
+    # Env variables used by CMakeCross.txt to select the desired
+    # crossbuild-essential-<arch> toolchain.
+    pkg.environment('DEB_HOST_ARCH', platform.architecture)
+    pkg.environment('DEB_HOST_GNU_TYPE', platform.platform_triple)
+    toolchain = '-DCMAKE_TOOLCHAIN_FILE=/etc/dpkg-cross/cmake/CMakeCross.txt'
+
+    boost_static_flag = "-DBOOST_STATIC=OFF"
+    yamlcpp_static_flag = "-DYAMLCPP_STATIC=OFF"
+    special_flags += ["-DBOOST_INCLUDEDIR=/usr/include",
+                      "-DBOOST_LIBRARYDIR=/usr/lib/#{platform.platform_triple}",
+                      "-DCMAKE_FIND_ROOT_PATH='/opt/puppetlabs/puppet;/usr'"].join(' ')
+
+    special_flags += ' -DENABLE_CXX_WERROR=OFF' if platform.name =~ /debian-10/
   elsif platform.is_cross_compiled_linux?
     ruby = "#{settings[:host_ruby]} -r#{settings[:datadir]}/doc/rbconfig-#{settings[:ruby_version]}-orig.rb"
     toolchain = "-DCMAKE_TOOLCHAIN_FILE=/opt/pl-build-tools/#{settings[:platform_triple]}/pl-build-toolchain.cmake"
-- 
2.25.0

